campaign-launcher/interfaceV2/src/api/httpClient.ts

```diff
@@ -0,0 +1,124 @@
+import axios, { 
+  AxiosError, 
+  AxiosInstance, 
+  AxiosRequestConfig, 
+  AxiosRequestHeaders, 
+  AxiosResponse, 
+  InternalAxiosRequestConfig 
+} from 'axios';
+
+import { tokenManager, TokenData } from '../utils/TokenManager';
+
+type HttpClientConfig = {
+  baseURL: string;
+  headers?: AxiosRequestHeaders;
+};
+
+type RefreshPromise = Promise<AxiosResponse<TokenData>>;
+
+export const REFRESH_FAILURE_EVENT = 'refresh-failure';
+
+export class HttpClient {
+  private axiosInstance: AxiosInstance;
+  private refreshPromise: RefreshPromise | null = null;
+
+  constructor(config: HttpClientConfig) {
+    this.axiosInstance = axios.create({
+      baseURL: config.baseURL,
+      headers: {
+        'Content-Type': 'application/json',
+        ...config.headers,
+      },
+    });
+
+    this.setupInterceptors();
+  }
+
+  private setupInterceptors(): void {
+    this.axiosInstance.interceptors.request.use(
+      (config: InternalAxiosRequestConfig) => {
+        const accessToken = tokenManager.getAccessToken();
```

Comment:
Token manager shouldn't be used directly, it should be injected as a dependency instead